win32/service.py

```diff
@@ -0,0 +1,183 @@
+"""
+A Windows Service wrapper for win32/wsupervisor.py which consists in a minimalistic
+supervisor for our agent with restart tries in case of failure. This program
+will be packaged into an .exe file with py2exe in our omnibus build. It doesn't
+have any project dependencies and shouldn't. It just launches and talks to our
+Windows supervisor in our Python env. This way the agent code isn't shipped in
+an .exe file which allows for easy hacking on it's source code.
+
+Many thanks to the author of the article below which saved me quite some time:
+http://ryrobes.com/python/running-python-scripts-as-a-windows-service/
+
+"""
+# stdlib
+import os
+import socket
+import select
+import logging
+
+# 3p
+import psutil
+import win32api
+import subprocess
+import win32event
+import win32service
+import win32serviceutil
+import servicemanager
+
+import ctypes
+from ctypes import wintypes, windll
+
+
+def _windows_commondata_path():
+    """Return the common appdata path, using ctypes
+    From http://stackoverflow.com/questions/626796/\
+    how-do-i-find-the-windows-common-application-data-folder-using-python
+    """
+    CSIDL_COMMON_APPDATA = 35
+
+    _SHGetFolderPath = windll.shell32.SHGetFolderPathW
+    _SHGetFolderPath.argtypes = [wintypes.HWND,
+                                 ctypes.c_int,
+                                 wintypes.HANDLE,
+                                 wintypes.DWORD, wintypes.LPCWSTR]
+
+    path_buf = wintypes.create_unicode_buffer(wintypes.MAX_PATH)
+    _SHGetFolderPath(0, CSIDL_COMMON_APPDATA, 0, 0, path_buf)
+    return path_buf.value
+
+
+# Let's configure logging accordingly now (we need the above function for that)
+logging.basicConfig(
+    filename=os.path.join(_windows_commondata_path(), 'Datadog', 'logs', 'service.log'),
+    level=logging.DEBUG,
```

Comment:
Yeah, that an issue. We could just not display it in the GUI but only send it with flares.